app/_components/SubmitGuessButton.tsx

```diff
@@ -0,0 +1,111 @@
+import { useState, useEffect } from "react";
+import { Button } from "./ui/button";
+import { Pin } from "../_utils/global";
+import {
+  Coordinates,
+  GetDistanceFromCoordinatesToMeters,
+} from "../_utils/coordinateMath";
+
+
+interface SubmitGuessButtonProps {
+  pins: Pin[];
+}
+
+/**
+ * TODO: Manage Button states based on closest pin's search radius.
+ * This can done by constantly checking the user coordinates.
+ * Manage how frequent this call is made.
+ **/
+
+function SubmitGuessButton({
+  pins,
+}: SubmitGuessButtonProps): React.JSX.Element {
+  const [trackingPin, setTrackingPin] = useState<Pin | null>(null);
+  const [distanceToPin, setDistanceToPin] = useState<number>(0);
+  const [isActiveState, setIsActiveState] = useState<boolean>(false);
+
+  useEffect(() => {
+    const id = navigator.geolocation.watchPosition((position) => {
+      handleTrackingPinAndDistanceToPin(position.coords);
+    });
+    return () => navigator.geolocation.clearWatch(id);
+  });
+
+  useEffect(() => {
+    if (!trackingPin) return;
+    setIsActiveState(isWithinSearchZone());
+  }, [trackingPin, distanceToPin]);
+
+
+  const handleTrackingPinAndDistanceToPin = (
+    userCoords: GeolocationCoordinates
+  ) => {
+    const userCoordinates: Coordinates = {
+      longitude: userCoords.longitude,
+      latitude: userCoords.latitude,
+    };
+    let shortestDistance: number = Number.MAX_SAFE_INTEGER;
+    let pinToTrack: Pin | null = null;
+
+    //Finds the closest pin
+    for (const pin of pins) {
+      //Change based on new schema
+      if(pin.collect){
+        continue;
+      }
+      const pinCoordinates: Coordinates = {
+        longitude: pin.longitude,
+        latitude: pin.latitude,
+      };
+      const distance: number = GetDistanceFromCoordinatesToMeters(
+        userCoordinates,
+        pinCoordinates
+      );
+      if (distance < shortestDistance) {
+        shortestDistance = distance;
+        pinToTrack = pin;
+      }
+    }
+    setTrackingPin(pinToTrack);
+    setDistanceToPin(shortestDistance);
+  };
```

Comment:
There's a lot going on here, making it hard to read. It'd be ideal to break this down into separate, reusable functions instead.

It also seems to be relying on variables or props that are outside of its scope. What happens if the prop `pins` is undefined?